lib/openhab/core/items/item.rb

```diff
@@ -28,6 +28,25 @@ def ===(other)
         #   The item's descriptive label.
         #   @return [String, nil]
 
+        #
+        # @!attribute [rw] tags
+        #   The item's tags
+        #   @return [Array<String>]
+        #   @overload tags
+        #     Returns the item's tags.
+        #     @return [Array<String>]
+        #   @overload tags=(*tags)
+        #     Sets the item's tags
+        #     @param [String,Symbol,Semantics::Tag] tags Tags to set.
+        #     @return [void]
+
+        # @!visibility private
+        def tags=(*tags)
+          remove_all_tags
+          tags = DSL::Items::ItemBuilder.normalize_tags(tags.flatten)
+          add_tags(tags)
```

Comment:
after this method is complete, you need to call `Provider.registry.provider_for(self).update(self)`, otherwise a) your changes won't be persisted, and b) nothing else in the rest of openHAB will notice that tags changed if they need to recompute stuff.